cpp/log/etcd_consistent_store-inl.h

```diff
@@ -686,18 +670,21 @@ util::Status EtcdConsistentStore<Logged>::CleanupOldEntries() {
     }
     VLOG(1) << "Cleanup deleted unsequenced entry (" << util::ToBase64(hash)
             << ") corresponding to sequenced entry #" << sequence_number;
-
-    // Now delete the sequenced entry:
-    status = DeleteEntry(&entry);
-    if (!status.ok()) {
-      LOG(WARNING) << "Cleanup couldn't delete sequenced entry #"
-                   << sequence_number << " (but corresponding unsequenced "
-                   << "entry was successfully deleted) : " << status;
-      return status;
-    }
-    VLOG(1) << "Cleanup deleted sequenced entry #" << sequence_number;
   }
-  return util::Status::OK;
+  // Now remove mappings for deleted sequence numbers.
+  size_t to, from(i);
+  for (to = 0; from < sequence_mapping.Entry().mapping_size(); ++from, ++to) {
+    sequence_mapping.MutableEntry()->mutable_mapping()->SwapElements(from, to);
```

Comment:
Other than the logging (that I'd expect to be a bit noisy, actually, maybe collapse it into a single message with the first and last sequence numbers?), why not ```DeleteSubrange``` directly?